web/app/scripts/views/map/map-controller.js

```diff
@@ -88,6 +90,11 @@
                 params.polygon = userDrawnPoly;
             }
 
+            // Get a tilekey for the CSV export
+            QueryBuilder.djangoQuery(true, 0, _.extend({ limit: 0 }, params)).then(function(records) {
```

Comment:
I haven't checked, but I'm assuming the back-end is able to optimize the limit=0 query to be pretty easy.  But yeah, both points make sense.  I moved the tilekey query into the export call.